cmd/passthrough.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+)
+
+type PassthroughCommand struct {
+	bin  string
+	name string
+	args []string
+}
+
+func NewPassthroughCommand(name string, bin string, args []string) *PassthroughCommand {
+	return &PassthroughCommand{
+		name: name,
+		bin:  bin,
+		args: args,
+	}
+}
+
+// Taken from https://github.com/kubernetes/kubectl/blob/b155278f1f4a21a0be2d4f6f0037258dee4d1a22/pkg/cmd/cmd.go#L371
+func (c *PassthroughCommand) Run(args []string) int {
+	// Windows does not support exec syscall.
+	if runtime.GOOS == "windows" {
+		cmd := exec.Command(c.bin, args...)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Stdin = os.Stdin
+		cmd.Env = os.Environ()
+		err := cmd.Run()
+		if err == nil {
+			return 0
+		}
+		return 1
+	}
+
+	// invoke cmd binary relaying the environment and args given
+	// append executablePath to cmdArgs, as execve will make first argument the "binary name".
+	if err := syscall.Exec(c.bin, append([]string{c.bin}, args...), os.Environ()); err != nil {
```

Comment:
Do you know if this method is preferred to the windows "workaround" above?

If the windows method works fine, couldn't we just use that for all platforms unconditionally?

Reply:
See: https://github.com/kubernetes/kubernetes/pull/76227#discussion_r277102122

I must admit I don't understand the implications of "replaces the old process with the new one"

Reply:
Same. Well I know enough from looking into it before that there is a difference. Still not sure if it matters here but might as well stick with what they have 👍